### PR TITLE
refactor: use impl From for datafusion error

### PR DESCRIPTION
**Pull request by bokuweb**
refactor: datafusion error. we might use `impl From` for error conversion.

### CODE REVIEW

File: src/common/infra/errors.rs [hunk 1]
```diff
@@ -35,8 +35,6 @@ pub enum Error {
     SimdJsonError(#[from] simd_json::Error),
     #[error("ArrowError# {0}")]
     ArrowError(#[from] datafusion::arrow::error::ArrowError),
-    #[error("DataFusionError# {0}")]
-    DataFusionError(#[from] datafusion::error::DataFusionError),
```
Comment (bokuweb) [inline src/common/infra/errors.rs:39]: This field is not used.

File: src/service/search/grpc/errors.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+use datafusion::{common::SchemaError, error::DataFusionError};
```
Comment (hengfeiyang) [inline src/service/search/grpc/errors.rs:1]: Need add license header at the top.

File: src/service/search/grpc/mod.rs [hunk 3]
```diff
@@ -12,14 +12,11 @@
 //
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
+mod errors;
```
Comment (hengfeiyang) [inline src/service/search/grpc/mod.rs:15]: Need add a black line after license header.

**Review by hengfeiyang — CHANGES_REQUESTED** [review]

File: src/service/search/grpc/errors.rs (hunk 2 above)
Comment (bokuweb) [reply src/service/search/grpc/errors.rs:16]: fixed

File: src/service/search/grpc/mod.rs (hunk 3 above)
Comment (bokuweb) [reply src/service/search/grpc/mod.rs:16]: ditto

**Review by hengfeiyang — APPROVED** [review]

**Pull request merged**